airflow/cli/cli_parser.py

```diff
@@ -58,9 +58,15 @@ class DefaultHelpParser(argparse.ArgumentParser):
     def _check_value(self, action, value):
         """Override _check_value and check conditionally added command"""
         executor = conf.get('core', 'EXECUTOR')
-        if value == 'celery' and executor not in (CELERY_EXECUTOR, CELERY_KUBERNETES_EXECUTOR):
-            message = f'celery subcommand works only with CeleryExecutor, your current executor: {executor}'
-            raise ArgumentError(action, message)
+        if value == 'celery':
+            if executor != CELERY_EXECUTOR:
```

Comment:
This does not cover `CELERY_KUBERNETES_EXECUTOR` though, right?

Reply:
i.e. if we don't run it via the Helm Chart, it should still work so I don't think this change is needed, apart from improving the error message

Reply:
@kaxil  the problem is that when executor==CELERY_KUBERNETES_EXECUTOR, `airflow celery worker` does not work

you get some fork pool error

running `airflow celery worker` with `CELERY_EXECUTOR` instead resolves the issue

celery workers don't need to know that _the scheduler_ is using CKE

i know this is a bit hacky.  our code should handle having all components set to use the same executor.  however, that simply doesn't work right now, and this change (forcing the C workers to think CeleryExecutor is used) makes CKE work again immediately.

perhaps it makes sense to push out this hacky fix and then later look into why we get fork pool issues.  but that's a question for you guys

Reply:
Yea that one feels too hacky, we should at least take care of it when running "airflow celery worker" (a less hacky --- so that users don't need to have a different configuration for the Celery Worker.

Can you paste the full stack trace of the error please too -- want to see if I have seen that before

Reply:
See issue #13263

Reply:
> we should at least take care of it when running "airflow celery worker" (a less hacky --- so that users don't need to have a different configuration for the Celery Worker.

another way to think about it @kaxil is, why does the celery worker even need _any_ executor configured?

the celery worker does what the celery worker does.  it shouldn't really need to instantiate an executor at all i would think.  shouldn't only the scheduler need to do that?

so while presumably we can fix this fork issue by making changes to CKE like you were doing in your local branch, another possible approach might be to make it so that the celery worker command somehow doesn't create / use / invoke an executor.